admin/failures/overview/endpoint.js

```diff
@@ -28,9 +37,10 @@ module.exports = (purchases) => (req, res) => {
             })
             .map((o) => {
                 const latestError = o.data.errors && o.data.errors.length && o.data.errors[o.data.errors.length - 1];

```

Comment:
This would extract nicely into a function `getLastElement(o.data.errors)`, in which you spell out the if-statements and return early to make the code super clear:

```js
function getLastELement(list) {
  if(!list || !list.length) {
    return undefined;
  }
  return list[list.length - 1];
}
```

See how easy that is to skim and understand??  :smile: